src/Microsoft.Health.Dicom.Api/Registration/DicomServerServiceCollectionExtensions.cs

```diff
@@ -95,7 +95,10 @@ public static IDicomServerBuilder AddBackgroundWorkers(this IDicomServerBuilder
                     options.RespectBrowserAcceptHeader = true;
                     options.OutputFormatters.Insert(0, new DicomJsonOutputFormatter());
                 })
-                .AddJsonSerializerOptions(o => o.Converters.Add(new JsonStringEnumConverter()));
+                //
+                //  JsonStringEnumConverter doesn't provide good enough error message, make custom one to fit our requirements.
+                //  CustomJsonStringEnumConverter is used for extended query tag Apis.
+                .AddJsonSerializerOptions(o => o.Converters.Add(new CustomJsonStringEnumConverter()));
```

Comment:
Is the default error message really that bad? For parsing enums, it appears to be like `"The value '<value>' is not valid for <Type>."` when I test it locally. I'd definitely prefer not to rewrite the enum parsing simply because of a less-than-ideal error message. If we want to change the error message, I think we can configure the `ModelBindingMessageProvider` when configuring `AddMvc`

Reply:
* There are 2 reason the `JsonStringEnumConverter` not fit well in our case. 

1.  Error message is good for developer, but not for end user.   Check the error message below, we know what is `[Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag.QueryStatus]`, but custom don't know this, and no idea how to fix it -- we need to give valid value sets.
<img src="https://user-images.githubusercontent.com/67345894/134748497-07731416-f1d1-4229-91f4-828d181ea554.png" width="500" />

**Vs**. improved error message:
<img src="https://user-images.githubusercontent.com/67345894/134748668-a307d50c-d74f-40bd-b36a-79eb6442a36d.png" width="500" />

2.  `JsonStringEnumConverter` treat "1" as string, so accept as valid enum value, which I'm trying to block -- the number value is what we use internally, don't want to expose and confuse customer (even setting `allowIntegerValues` to `false` cannot block it)

* I explored `ModelBindingMessageProvider`, but seems not work in our case, best guess is the error comes from `JsonDeserializer` where model binding doesn't have enough insight --- I override accessor for every message in `ModelBindingMessageProvider` like below, but they don't show in error message.
<img src="https://user-images.githubusercontent.com/67345894/134749166-4c58add2-c5e7-4961-ae3d-f93a125a1f77.png" width="800" />

Reply:
Maybe I should not block number  string (e.g"1"), that's default behavior of `Enum.TryParse`

Reply:
I think it's a bit strange. I am not wholly opposed to leaving that functionality in for now and we can remove it from our API later on. We just won't document it that behavior.

Reply:
If we decide to leave it in for now, I'd say remove the custom converter until we remove the functionality.

Reply:
As discussed, will only use custom converter for our own Enum `QueryStatus`